src/site/content/en/secure/securely-hosting-user-data/index.md

```diff
@@ -0,0 +1,103 @@
+date: 2023-06-01
+description: >
+  How to search for vulnerabilities on your site.
+tags:
+  - security
+---
+
+Many web applications need to display user-controlled content. This can be as simple as serving user-uploaded images (for example, profile photos), or as complex as rendering user-controlled HTML (for example, a web development tutorial). This has always been difficult to do securely, so we've worked to find easy, but secure solutions that can be applied to most types of web applications.
+
+## Classical Solutions for Isolating Untrusted Content
+
+The classic solution for securely serving user-controlled content is to use what are known as ["sandbox domains''](https://security.googleblog.com/2012/08/content-hosting-for-modern-web.html). The basic idea is that if your application's main domain is `example.com`, you could serve all untrusted content on `exampleusercontent.com`. Since these two domains are [cross-site](/same-site-same-origin), any malicious content on `exampleusercontent.com` can't impact `example.com`.  
+This approach can be used to safely serve all kinds of untrusted content including images, downloads, and HTML. While it may not seem like it is necessary to use this for images or downloads, doing so helps avoid risks from [content sniffing](https://en.wikipedia.org/wiki/Content_sniffing), especially in legacy browsers.  
+Sandbox domains are widely used across the industry and have worked well for a long time. But, they have two major downsides:
+
+1. Applications often need to restrict content access to a single user, which requires implementing authentication and authorization. Since sandbox domains purposefully do not share cookies with the main application domain, this is very difficult to do securely. To support authentication, sites either have to rely on [capability URLs](https://www.w3.org/TR/capability-urls), or they have to set separate authentication cookies for the sandbox domain. This second method is especially problematic in the modern web where many browsers restrict cross-site cookies by default.
+1. While user content is isolated from the main site, it isn't isolated from other user content. This creates the risk of malicious user content attacking other data on the sandbox domain (e.g. via reading same-origin data).
+
+It is also worth noting that sandbox domains help mitigate phishing risks since resources are clearly segmented onto an isolated domain.
+
+## Modern Solutions for Serving User Content
+
+Over time the web has evolved, and there are now easier, more secure ways to serve untrusted content. There are many different approaches here, so we will outline two solutions that are currently in wide use at Google.
+
+### Approach 1: Serving Inactive User Content
+
+If a site only needs to serve inactive user content (i.e. content that is not HTML/JS, for example images and downloads), this can now be safely done without an isolated sandbox domain. There are two key steps:
```

Comment:
```suggestion
If a site only needs to serve inactive user content (that is content that is not HTML or JavaScript, for example images and downloads), this can now be safely done without an isolated sandbox domain. There are two key steps:
```